src/Samples/Sarif.WorkItems.Sample/ProgramStubs.cs

```diff
@@ -0,0 +1,233 @@
+
+            SarifLog matchedLog = new SarifLog
+            {
+                Runs = new[]
+                {
+                    resultMatcher.CreateBaselinedRun(baselineSarif.Runs[0], currentSarif.Runs[0])
+                }
+            };
+
+            // 3. Make sure everything has a GUID and that we have a map back to each result from it
+            var guidToMatchedResultMap = new Dictionary<string, Result>();
+            foreach (Result result in matchedLog.Runs[0].Results)
+            {
+                if (result.Guid == null)
+                {
+                    result.Guid = Guid.NewGuid().ToString();
+                }
+                guidToMatchedResultMap[result.Guid] = result;
+            }
+
+            var adoClient = new AzureDevOpsClient();
+            var filingTarget = new AzureDevOpsFilingTarget(adoClient);
+
+            var workItemFiler = new WorkItemFiler(
+                filingTarget,
+                filteringStrategy: new NewResultsFilteringStrategy(),
+                groupingStrategy: new OneResultPerWorkItemGroupingStrategy());
+
+            string logFilePath = Path.GetTempFileName();
+            File.WriteAllText(logFilePath, JsonConvert.SerializeObject(matchedLog));
+
+            // IMPORTANT! There's no facility defined yet for creating/configuring titles, area paths, etc.
+
+            try
+            {
+                IEnumerable<ResultGroup> resultGroups = workItemFiler.FileWorkItems(logFilePath).Result;
+                foreach (ResultGroup resultGroup in resultGroups)
+                {
+                    foreach (Result result in resultGroup.Results)
+                    {
+                        bool matched = guidToMatchedResultMap.TryGetValue(result.Guid, out Result originalResult);
+                        Debug.Assert(matched);
+                        originalResult.WorkItemUris = result.WorkItemUris;
+                    }
+                }
+            }
+            finally
+            {
+                if (File.Exists(logFilePath)) File.Delete(logFilePath);
+            }
+        }
+#endif
+
+        #region Stubs
+        private static SarifLog RetrieveCurrentSarif(string v)
+        {
+            string sarifText = GetResourceText("Sarif.Sdk.Sample.SampleTestFiles.Current.sarif");
+            return JsonConvert.DeserializeObject<SarifLog>(sarifText);
+        }
+
+        private static SarifLog RetrieveBaselineSarif(string v)
+        {
+            string[] resourceNames = typeof(Program).Assembly.GetManifestResourceNames();
+            Console.WriteLine(String.Join("\r\n", resourceNames));
+
+            string sarifText = GetResourceText("Sarif.Sdk.Sample.SampleTestFiles.Baseline.sarif");
+            return JsonConvert.DeserializeObject<SarifLog>(sarifText);
+        }
+
+        public static string GetResourceText(string resourceName)
```

Comment:
>GetResourceText [](start = 29, length = 15)

FYI Scott wrote a nice `ResourceExtractor` class.

Reply:
thanks for the information. none of the program stubs code is 'live', this is all throwaway.

---
In reply to: [307081180](https://github.com/microsoft/sarif-sdk/pull/1589#discussion_r307081180) [](ancestors = 307081180)